hyper-v: Added 4 Stress TC

asofron opened the pull request.

chvalean commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,210 @@
+#       No optional parameters needed
+#
+########################################################################
+CONSTANTS_FILE="constants.sh"
```
these 2 variables should not be required

asofron replied:
Remove Constants variable

chvalean commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,210 @@
+function Cpu-Test ()
+{
+    LogMsg "Creating cpu.log and starting test."
+    sysbench cpu --num-threads=1 run > /home/lisa/cpu.log
```
don't hardcore path

asofron replied:
Created path variable

chvalean commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,210 @@
+    tar xvfvz autoconf-2.69.tar.gz
+    cd autoconf-2.69
+    ./configure
+    make
```
make && make install, single line

asofron replied:
made to one line inside function

chvalean commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,210 @@
+    make
+    make install
+
+    yum install devtoolset-2-binutils -y
```
there should be an existing function to install the packages

asofron replied:
Created function for dependencies and sysbench install

chvalean commented on Testscripts/Windows/ChangeCPU.ps1:
```diff
@@ -0,0 +1,134 @@
+########################################################################
```
Wasn't there a single script to handle cpu and/or memory changes?

asofron replied:
Removed duplicate testcase

chvalean commented on Testscripts/Windows/STRESS-mtu-netvsc-reload.ps1:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    Start-Sleep -s 600
+    "${vmName} IP Address after reloading hv_netvsc: ${Ipv4}"
```
I don't see where the IP of VM is checked if it's still matching or it changed

asofron replied:
Added IP Verfication portion

chvalean commented on XML/TestCases/StressTests.xml:
```diff
@@ -43,4 +43,75 @@
+        <Tags>stress</Tags>
+    </test>
+    <test>
+        <testName>STRESS-RELOAD-MODULES-UP</testName>
```
this TC should use the same files from existing TC STRESSTEST-RELOAD-MODULES-UP

asofron replied:
Removed duplicate testcase

adriansuhov commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,210 @@
+else
+    LogMsg "Error: no ${CONSTANTS_FILE} file"
+    SetTestStateAborted
+    exit 10
```
replace all exit codes with "exit 0". lisav2 doesn't handle so well exit codes !=0.

asofron replied:
Fixed

adriansuhov commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,210 @@
+elif is_ubuntu ; then
+    apt-get install automake -y
+    apt-get install libtool -y
+    apt-get install pkg-config -y
```
use a single liner (apt-get install pack1 pack2 pack3 -y )

adriansuhov commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,210 @@
+    make install
+
+    yum install devtoolset-2-binutils -y
+    yum install automake -y
```
use single liner to install stuff

asofron replied:
made it one line

adriansuhov commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,210 @@
+    yum install libtool -y
+    yum install vim -y
+
+    pushd "$ROOT_DIR/sysbench-$Sysbench_Version"
```
I see that sysbench install is the same on all 3 distros (pushd/bash/make/make install/exit code check). Can you make a function to avoid duplicate code?

adriansuhov commented on Testscripts/Windows/DM-CONFIGURE-MEMORY.ps1:
```diff
@@ -0,0 +1,250 @@
+                    Start-sleep -s 5
+                    $timeout = $timeout - 5
+                }
+
```
example of unnecessary new line . CLosing brackets shouldn't have new lines between them.

asofron replied:
removed

adriansuhov commented on Testscripts/Windows/DM-CONFIGURE-MEMORY.ps1:
```diff
@@ -0,0 +1,250 @@
+
+               LogMsg "Stopping VM $VMName"
+                Stop-VM -Name $VMName -ComputerName $HvServer -force
+
```
example of unnecessary new line. we have a command and then a check so they are related. No need for a new line in this case

asofron replied:
removed

adriansuhov commented on Testscripts/Windows/Multiple-Reboot.ps1:
```diff
@@ -0,0 +1,160 @@
+        Restart-VM -VMName $VMName -ComputerName $HvServer -Force
+
+        Start-Sleep 5
+
```
check unnecessary new lines in this file too

asofron replied:
removed

adriansuhov commented on Testscripts/Windows/STRESS-BootLargeMemory.ps1:
```diff
@@ -0,0 +1,91 @@
+    #	Main body script
+    #
+    #######################################################################
+
```
double new lines

asofron replied:
removed

chvalean commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,167 @@
+    LogMsg "Preparing files to test $1..."
+    sysbench fileio --num-threads=1 --file-test-mode=$1 run >${ROOT_DIR}/$1.log
+    if [ $? -ne 0 ]; then
+        LogMsg "ERROR: Unable to execute sysbench fileio mode $1. Aborting..."
```
logerr and remove error:

chvalean commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,167 @@
+    LogMsg "Cloning sysbench"
+    wget https://github.com/akopytov/sysbench/archive/$Sysbench_Version.zip
+    if [ $? -gt 0 ]; then
+        LogMsg "Failed to download sysbench."
```
logerr

chvalean commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,167 @@
+    case $DISTRO in
+    fedora*)
+        pushd $ROOT_DIR
+        mkdir autoconf
```
remove mkdir and cd lines, it's fine to have homedir/autoconf-2.69 folder

chvalean commented on Testscripts/Linux/Sysbench.sh:
```diff
@@ -0,0 +1,167 @@
+    LogMsg "Installing sysbench"
+    make && make install
+    if [ $? -ne 0 ]; then
+        LogMsg "ERROR: Unable to install sysbench. Aborting..."
```
logerr

chvalean commented on Testscripts/Windows/Multiple-Reboot.ps1:
```diff
@@ -0,0 +1,142 @@
+#     This script will reboot a VM as many times as specified in the count parameter
+#     and check that the VM reboots successfully.
+#
+#     The ICA scripts will always pass the vmName, hvServer, and a
```
remove lines 11 to 17

chvalean commented on Testscripts/Windows/RevertSnapshot.ps1:
```diff
@@ -0,0 +1,100 @@
+        LogErr "hvServer is null"
+        return $False
+    }
+    if (-not $TestParams) {
```
remove this

Commit:
Added Stress Test cases

Commit:
Fix

- Fixed Sysbench.sh 
- Fixed STRESS-BootLargeMemory.ps1
- Fixed SETUP-DM-CONFIGURE-MEMORY.ps1
- Fixed Multiple-Reboot.ps1
- Run PS Script Analyzer
